EESSI-extend-easybuild.eb

```diff
@@ -196,10 +201,30 @@ elseif (project_modulepath ~= nil) then
   -- configure MODULEPATH
   prepend_path("MODULEPATH", project_modulepath)
 end
+
 -- Make sure EasyBuild itself is loaded
+-- need to also handle the unload behaviour where the version is defined only before we unload
+easybuild_version = os.getenv("EBVERSIONEASYBUILD")
 if not ( isloaded("EasyBuild") ) then
     load(latest("EasyBuild"))
 end
+easybuild_version = os.getenv("EBVERSIONEASYBUILD") or easybuild_version
+eessi_version = os.getenv("EESSI_VERSION") or "2023.06"
+
+-- Set environment variables that are EasyBuild version specific
+if convertToCanonical(easybuild_version) > convertToCanonical("4") then
+  setenv ("EASYBUILD_STRICT_RPATH_SANITY_CHECK", "1")
+  setenv ("EASYBUILD_CUDA_SANITY_CHECK_ERROR_ON_FAILED_CHECKS", "1")
+  setenv ("EASYBUILD_ENV_FOR_SHEBANG", "1")
+  setenv ("EASYBUILD_FAIL_ON_MOD_FILES_GCCCORE", "1")
+  setenv ("EASYBUILD_LOCAL_VAR_NAMING_CHECK", "error")
+  -- Set environment variables that are EESSI version specific
+  if convertToCanonical(eessi_version) > convertToCanonical("2023.06") then
+    setenv ("EASYBUILD_PREFER_PYTHON_SEARCH_PATH", "EBPYTHONPREFIXES")
```

Comment:
I don't have strong feelings about this, and I do see the appeal to use `$EBPYTHONPREFIXES` rather than `$PYTHONPATH` (works better when using Python virtual environments), but this is in some sense also uncharted territory, so we're bound to run into surprises/problems because we're using this.

Also, not relying on the standard mechanism (`$PYTHONPATH`) is no doubt going to confuse end users at some point.

So, we should clearly motivate why we're going down this road, since we can't change our mind in the middle, this is an all-or-nothing choice...

Reply:
The motivation is exactly that, better `venv` behaviour, but I agree with the main concern about how hard it is to roll back. I was actually hoping you would be the one to sooth me.

End users can still use PYTHONPATH, but we wouldn't. Or are you more concerned about those using `EESSI-extend`?

Reply:
I'm mostly concerned about people being confused as to how Python packages can be imported from non-standard locations while we're not using `$PYTHONPATH`.

The `$EBPYTHONPREFIXES` approach seems robust, but it's very custom.

That said, I'm not opposing it, just making sure we're aware that we're (yet again) taking a non-standard route here, and there may be dragons along that route...

We should probably add some documentation on the different ways in which we divert from default EasyBuild configuration, and explain the motivation behind those settings => https://github.com/EESSI/docs/issues/519

Reply:
`EBPYTHONPREFIXES` isn't the default? I thought we would change that for EB 5